lib/std/fs/file.zig

```diff
@@ -314,6 +314,58 @@ pub const File = struct {
         mtime: i128,
         /// Creation time in nanoseconds, relative to UTC 1970-01-01.
         ctime: i128,
+
+        const Self = @This();
+
+        pub fn system_stat_kind_to_fs_kind(st: os.system.Stat) Kind {
+            const kind: File.Kind = if (builtin.os.tag == .wasi and !builtin.link_libc)
+                switch (st.filetype) {
+                    .BLOCK_DEVICE => File.Kind.BlockDevice,
+                    .CHARACTER_DEVICE => File.Kind.CharacterDevice,
+                    .DIRECTORY => File.Kind.Directory,
+                    .SYMBOLIC_LINK => File.Kind.SymLink,
+                    .REGULAR_FILE => File.Kind.File,
+                    .SOCKET_STREAM, .SOCKET_DGRAM => File.Kind.UnixDomainSocket,
```

Comment:
Why is NamedPipe not possible?
How did you rule out the other three file kinds?
```txt
        Whiteout,
        Door,
        EventPort,
```

Reply:
I didn't quite write this code, so I don't know. It's a copy of the [code (from line 401 to 428)](https://github.com/ziglang/zig/blob/e54ab5ebd02da5eaec97e79add5df5ebca8142af/lib/std/fs/file.zig#L401). I wanted to keep the same behavior, so I didn't mind too much what was in there.

Also: it would be probably best to de-duplicate this code, once everything else is in order.